packages/gitbook-v2/src/middleware.ts

```diff
@@ -121,8 +122,13 @@ async function serveSiteRoutes(requestURL: URL, request: NextRequest) {
                 apiToken,
             })
         );
-        const cookies: ResponseCookies = [];
 
+        //
+        // Retrieve the response cookie for visitor params that were passed to the URL to persist them in a session cookie.
+        //
+        const visitorParamsCookie = getResponseCookieForVisitorParams(unsignedClaims);
+
```

Comment:
Why not return it from `getVisitorPayload`? We should avoid too many visitor methods

Reply:
```js
const {visitorToken, visitorPuvlicCookie, visitorUnsignedClaims} = getVisitorPayload();
```

Reply:
Seperation of concerns. It's also following the same pattern that we use for the visitor auth token. One to perform the lookup and another one to retrieve the cookie response. Not too sure how is it problematic to have 2 seperate methods?

Reply:
it's also odd imo to have a function called `getVisitorPayload` return a cookie.

Reply:
Visitor auth is a bit different because setting the cookie for it is not very straight forward. You actually need the `basePath` that comes after the resolution API call hence separate function for it.

My intention with the suggestion was that `getVisitorPayload` (or maybe `getVisitorData`?) should be an API that should output all relevant info related to a visitor for a particular request.

Reply:
Ok I see your point on the API. Changing it.